pandas/plotting/_matplotlib/hist.py

```diff
@@ -243,15 +244,27 @@ def _grouped_hist(
     sharey : bool, default False
     rot : int, default 90
     grid : bool, default True
+    legend: : bool, default False
     kwargs : dict, keyword arguments passed to matplotlib.Axes.hist
 
     Returns
     -------
     collection of Matplotlib Axes
     """
 
+    if legend and "label" not in kwargs:
```

Comment:
Is there a reason why someone would pass `legend` and `label` together? Should this not just raise?

Reply:
The only reason I think is to rename for output:

````
index = 3 * ['1'] + 3 * ['2']
df = DataFrame([(3, 4)]*3 + [(5, 6)]*3, index=index, columns=['a', 'b'])
df.index.names = ['c']
ret = df.hist(by='c', legend=True, label=['Nice Name 1', 'Nice Name 2'])
````

![image](https://user-images.githubusercontent.com/45562402/82598553-d7601300-9b78-11ea-991b-1258432840a1.png)

Reply:
I think this should just raise instead

Reply:
It seems to me that we would then be raising on the only case where label has any effect - or am I missing another case?

Reply:
Also, I don't understand the reason why we would raise here when not raising makes it easier for the end user have different names appear in the legend. What does raising gain?

Reply:
So my thought is that If a user wants particular labels they should just provide that to the grouper; adding this here is just another way of doing things so just API clout

Reply:
I would completely agree if this was implemented within pandas. However it's being passed through to matplotlib via kwargs. Disabling certain kwargs from working and allowing others to go through seems hazardous to user expectations. But even if this is unconvincing, I think we should strive to be consistent with various plots:

````
index = 3 * ['1'] + 3 * ['2']
df = DataFrame([(3, 4)]*3 + [(5, 6)]*3, index=index, columns=['a', 'b'])
df.index.names = ['c']
ret = df.plot(y=['a', 'b'], label=['Nice Name 1', 'Nice Name 2'])
````

![image](https://user-images.githubusercontent.com/45562402/83193005-5965b400-a104-11ea-97c1-1da9957fa208.png)

Perhaps an issue could be raised as to how labels should be treated across all plots?

Reply:
Well shoot. As a counter-argument to what I just said. DataFrame.plot.hist ignores label altogether. I suppose if there is one plotting function we should be consistent with here, it's that one. What do you think about not raising, but simply ignoring any label kwarg?

Reply:
I still think we should raise instead of ignoring; ignoring leads to confusion or surprising behavior, so always better to be explicit